File: Python/array_second_largest.py
#Second largest element
arr = list(map(int, input("Enter elements: ").split()))
first = second = float('-inf')
for x in arr:
    if x > first:
        second = first
        first = x
    elif x > second and x != first:
        second = x
print("Second largest element:", second)

File: CPP/array_second_largest.cpp
//Find Second Largest Element
#include <iostream>
#include <climits>
using namespace std;

int main() {
    int n;
    cout << "Enter number of elements: ";
    cin >> n;
    int arr[n];

    cout << "Enter array elements: ";
    for (int i = 0; i < n; i++)
        cin >> arr[i];

    int first = INT_MIN, second = INT_MIN;
    for (int i = 0; i < n; i++) {
        if (arr[i] > first) {
            second = first;
            first = arr[i];
        } else if (arr[i] > second && arr[i] != first) {
            second = arr[i];
        }
    }

    cout << "Second largest element: " << second << endl;
    return 0;
}
